src/app/modules/service-providers/services/navigation/navigation.service.ts

```diff
@@ -1,16 +1,101 @@
 import { Injectable } from '@angular/core';
-import { Store } from '@ngrx/store';
+import { select, Store } from '@ngrx/store';
+import { lineSlice, lineString, nearestPointOnLine, point, length } from '@turf/turf';
+import {
+  currentPositionSelector, getNavState,
+  nextWaypointIndexSelector,
+  routeDetailsSelector,
+  routePointsSelector
+} from '../../../../store/nav.selectors';
+import { filter, tap, withLatestFrom } from 'rxjs/operators';
+import {
+  PlayVoiceWpNotificationAction,
+  SetDistanceToEndpointAction,
+  SetIsNextWpNotifiedAction,
+  SetNextWaypointDistanceAction,
+  SetNextWaypointIndexAction
+} from '../../../../store/nav.actions';
+
 
 @Injectable({
   providedIn: 'root'
 })
 export class NavigationService {
 
-  constructor(private store: Store<any>) { }
+  constructor(private store: Store<any>) {
+    this.calcNextWaypoints();
+  }
+
+
+  calcNextWaypoints() {
```

Comment:
Can and should be an observable instead of method (just like the rest of the application)